### DISCUSSION TITLE
Generate a maven property in pom.xml with custom extension and codestart

### DISCUSSION THREAD

**Discussion by jorgecastro05** [quarkusio/quarkus#30721] [Q&A] [answered]
Hello,
I'm developing one custom extension with a codestart, and I need to generate a maven property in pom.xml file,  this extension contains starter code and dependencies for working with jacoco and sonarqube, ex:

     <sonar.coverage.jacoco.xmlReportPaths>target/jacoco-report/jacoco.xml</sonar.coverage.jacoco.xmlReportPaths>

What configuration is required on extension or codestart to includde a specific property in pom.xml in the generated project?

Putting the property on parent extension pom.xml or runtime module doesn't work.

[a comment by an automated account is not shown]

**Comment by ia3andy** [comment]
@jorgecastro05 is this extension meant to be published or just for company or personal use? the problem I see if this is published is that this would make the codestart incompatible with the other buildtool (Gradle, ...)

**Reply by jorgecastro05** [reply]
It is just for company internal use, and the build tool has defined to be Maven, with Java language

**Comment by jorgecastro05** [comment, marked as answer]
Creating a pom.xml file in base path of codestart of runtime module works successfully, pom.xml contains the root node and the property, default strategy for merge pom files on codestarts is **smart-pom-merge** so this property is added to target pom file.

Example path inside runtime module:
runtime/src/main/codestarts/quarkus/my-codestart/base/pom.xml

**pom.xml**
```
<?xml version="1.0" encoding="UTF-8"?>
<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    <properties>
        <sonar.coverage.jacoco.xmlReportPaths>target/jacoco-report/jacoco.xml</sonar.coverage.jacoco.xmlReportPaths>
    </properties>
</project>
```
